source/material_model/viscoelastic_plastic.cc

```diff
@@ -0,0 +1,798 @@
+                                   "compositional fields representing these components must be named "
+                                   "and listed in a very specific format, which is designed to minimize "
+                                   "mislabeling stress tensor components as distinct 'compositional "
+                                   "rock types' (or vice versa). For 2D models, the first three "
+                                   "compositional fields must be labeled 'stress_xx', 'stress_yy' and 'stress_xy'. "
+                                   "In 3D, the first six compositional fields must be labeled 'stress_xx', "
+                                   "'stress_yy', 'stress_zz', 'stress_xy', 'stress_xz', 'stress_yz'. "
+                                   "\n\n "
+                                   "Combining this viscoelasticity implementation with non-linear viscous flow "
+                                   "and plasticity produces a constitutive relationship commonly referred to "
+                                   "as partial elastoviscoplastic (e.g., pEVP) in the geodynamics community. "
+                                   "While extensively discussed and applied within the geodynamics "
+                                   "literature, notable references include: "
+                                   "Moresi et al. (2003), J. Comp. Phys., v. 184, p. 476-497. "
+                                   "Gerya and Yuen (2007), Phys. Earth. Planet. Inter., v. 163, p. 83-105. "
+                                   "Gerya (2010), Introduction to Numerical Geodynamic Modeling. "
+                                   "Kaus (2010), Tectonophysics, v. 484, p. 36-47. "
+                                   "Choi et al. (2013), J. Geophys. Res., v. 118, p. 2429-2444. "
+                                   "Keller et al. (2013), Geophys. J. Int., v. 195, p. 1406-1442. "
+                                   "\n\n "
+                                   "The overview below directly follows Moresi et al. (2003) eqns. 23-32. "
+                                   "However, an important distinction between this material model and "
+                                   "the studies above is the use of compositional fields, rather than "
+                                   "tracers, to track individual components of the viscoelastic stress "
+                                   "tensor. The material model will be updated when an option to track "
+                                   "and calculate viscoelastic stresses with tracers is implemented. "
+                                   "\n\n "
+                                   "Moresi et al. (2003) begins (eqn. 23) by writing the deviatoric "
+                                   "rate of deformation ($\\hat{D}$) as the sum of elastic "
+                                   "($\\hat{D_{e}}$) and viscous ($\\hat{D_{v}}$) components: "
+                                   "$\\hat{D} = \\hat{D_{e}} + \\hat{D_{v}}$.  "
```

Comment:
Can you include the plastic component?

Reply:
The documentation in `viscoelastic_plastic.h` and `viscoelastic_plastic.cc` were both updated to included the plastic component.